dali/ft/filecopy.cpp

```diff
@@ -2623,6 +2623,49 @@ void FileSprayer::setTarget(IDistributedFile * target)
     }
 }
 
+void FileSprayer::checkTargetPath(RemoteFilename & filename)
+{
+    StringBuffer targetFilePath;
+
+    filename.getRemotePath(targetFilePath);
+    // prune the leading "//" from remote path
+    targetFilePath.remove(0, 2);
+
+    StringBuffer netaddress;
+    filename.queryIP().getIpText(netaddress);
+    const char * pnetaddress = netaddress.str();
+
+    if (streq("127.0.0.1", pnetaddress))
+        throwError1(DFTERR_LocalhostAddressUsed, targetFilePath.str());
+
+    //remove ip address from the beginning of the path
+    targetFilePath.remove(0, netaddress.length());
+#ifdef _DEBUG
+    LOG(MCdebugInfo, unknownJob, "Target file path is '%s'", targetFilePath.str());
+#endif
+
+    const char * ptargetFilePath = targetFilePath.str();
+    const char pathSep = filename.getPathSeparator();
+    const char dotString[]    = {pathSep, '.', pathSep, '\0'};
+    const char dotDotString[] = {pathSep, '.', '.', pathSep, '\0'};
+
+    const char * isDotString = strstr(ptargetFilePath, dotString);
+    const char * isDotDotString = strstr(ptargetFilePath, dotDotString);
+    if ((isDotDotString != nullptr) || (isDotString != nullptr))
+        throwError3(DFTERR_InvalidTargetPath, ptargetFilePath, dotDotString, dotString);
```

Comment:
I wouldn't worry about cases where people "legally" use /.. / - there's no real benefit to supporting it and it's clearer if we don't